File: src/main/java/org/vladg/ast/ASTRequester.java
package org.vladg.ast;

import com.google.inject.Guice;
import lombok.extern.slf4j.Slf4j;
import org.eclipse.jdt.core.dom.CompilationUnit;
import org.eclipse.jdt.core.dom.FileASTRequestor;
import org.vladg.jafax.utils.inject.DependencyManager;

@Slf4j
public class ASTRequester extends FileASTRequestor {

    private final ASTVisitor astVisitor;

    public ASTRequester() {
        var injector = Guice.createInjector(new DependencyManager());
        this.astVisitor = injector.getInstance(ASTVisitor.class);
    }

    @Override
    public void acceptAST(String sourceFilePath, CompilationUnit ast) {
        log.info("Scanning file: " + sourceFilePath);
        astVisitor.setCurrentFileName(sourceFilePath);
        ast.accept(astVisitor);
    }

}

File: src/main/java/org/vladg/ast/ASTRequester.kt
package org.vladg.ast

import com.google.inject.Guice
import org.eclipse.jdt.core.dom.CompilationUnit
import org.eclipse.jdt.core.dom.FileASTRequestor
import org.vladg.jafax.utils.inject.DependencyManager
import org.vladg.logger

class ASTRequester : FileASTRequestor() {

    private val logger = logger()

    private val astVisitor: ASTVisitor

    init {
        val injector = Guice.createInjector(DependencyManager())
        astVisitor = injector.getInstance(ASTVisitor::class.java)
    }


    override fun acceptAST(sourceFilePath: String, ast: CompilationUnit) {
        logger.info("Scanning file: $sourceFilePath")
        ast.accept(astVisitor)
    }

}